pkg/api/v1/types.go

```diff
@@ -1051,15 +1051,22 @@ type ClusterStatus struct {
 	CCM ExternalCCMStatus `json:"ccm"`
 }
 
+type ExternalCCMMigrationStatus string
+
+var (
+	ExternalCCMMigrationNotNeeded   ExternalCCMMigrationStatus = "notNeeded"
+	ExternalCCMMigrationSupported   ExternalCCMMigrationStatus = "supported"
+	ExternalCCMMigrationUnsupported ExternalCCMMigrationStatus = "unsupported"
+	ExternalCCMMigrationInProgress  ExternalCCMMigrationStatus = "inProgress"
+)
+
 // ExternalCCMStatus represents the external CCM status
 // swagger:model ExternalCCMStatus
 type ExternalCCMStatus struct {
 	// ExternalCCM reflects the ClusterFeatureExternalCloudProvider in the cluster .spec.features
 	ExternalCCM bool `json:"externalCCM"`
```

Comment:
Do we need this field any longer? Previously, it was used to determine if the migration is needed, but now that's covered with the new status field.